### ISSUE TITLE
Feature Request: Export form a date to a date

### ISSUE THREAD

**Issue by fazlerabbi37** [brarcher/budget-watch#145]
It would be very helpful if the app had the capability of exporting csv from a date to a date. That would make the user capable of making a monthly or yearly csv export.

**Comment by brarcher** [comment]
I can add a new field to the exported CSV data which will show dates in a human readable format based on the user's locale. For example:
```
Nov 23, 2017 5:00:00 AM
```
This field would be ignored during imports, as the original date field would be used instead. Is this what you are looking for?

**Comment by brarcher** [comment]
Here is a build which includes the new formatted date field:

[budget-watch.apk.zip](https://github.com/brarcher/budget-watch/files/1690645/app-release.apk.zip)

Let me know if it is what you are looking for. If so, I can include the change in the next release.

**Comment by fazlerabbi37** [comment]
Thank you @brarcher for taking you time to make a build for the date feature which I think you should definitely include in the main build. I tested the build for 15 transactions and it works fine but unfortunately not the feature I wanted. You see if I want to make a monthly balance sheet from the exported csv which is possible, I have to make a export of today that  will contain all transactions till today then I will have to separate the transactions of the month I want the balance sheet for then transfer them to a separate sheet. Where it would be easier if the csv export can be done within a date range, ei. from January 1 to January 31. Thank you again for taking out you time to check out the feature request.

**Comment by brarcher** [comment]
Oh, my mistake. I though your request was related to your question earlier. (:

I think I understand your request. There is a similar function with the budget view, where the date range can be selected. Being able to pick the date range for exporting would also be useful.

Maybe it would look something like the following initially:

![image](https://user-images.githubusercontent.com/5264535/35762924-ad3c2d60-086e-11e8-8619-b82d16f33bfb.png)

Then when a date range is selected:

![image](https://user-images.githubusercontent.com/5264535/35762915-731b9594-086e-11e8-8041-1efc8c994efe.png)

What are your thoughts?

**Comment by fazlerabbi37** [comment]
Looks excellent mate ! :+1: Waiting eagerly to get the update.

**Comment by brarcher** [comment]
Oh, the pictures were just mock-ups. I do not think that it will be too difficult to do though, I'll just need to find some time. When I can get to it, I'll let you know here.

**Comment by brarcher** [comment]
I got some time this weekend to work on this. Attached is a build with the proposed feature. Would you mind taking a look to see if it is what you expect? Give me any feedback you may have.

[budget-watch.apk.zip](https://github.com/brarcher/budget-watch/files/1693308/app-release.apk.zip)

**Comment by fazlerabbi37** [comment]
So, I check the build today. It is exactly what I was looking for so thank
you very much. ​I am extremely sorry for not being able to reply earlier
though I saw your reply. Had a busy weekend and I was able to find some
time this evening and test out the app. Got time to make only 7
transactions and the export works fine. Thanks again. One question I have
thought, when are you planning to push these features in master breach, if
you are planning to do it at all?




<https://mailtrack.io/> Sent with Mailtrack
<https://chrome.google.com/webstore/detail/mailtrack-for-gmail-inbox/ndnaehgpjlnokgebbaldlmgkapkpjkkb?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality>

**Comment by fazlerabbi37** [comment]
So, I check the build today. It is exactly what I was looking for so thank you very much. ​I am extremely sorry for not being able to reply earlier though I saw your reply. Had a busy weekend and I was able to find some time this evening and test out the app. Got time to make only 7 transactions and the export works fine. Thanks again. One question I have thought, when are you planning to push these features in master breach, if you are planning to do it at all?
